python/ray/tests/test_actor_failures.py

```diff
@@ -1173,9 +1173,9 @@ class RestartableActor:
         def __init__(self):
             self.counter = 0
 
-        def sleep_and_increment(self, sleep_time, signal_actor):
-            ray.get(signal_actor.send.remote())
-            time.sleep(sleep_time)
+        def increment(self, signal_actor):
+            ray.get(signal_actor.send.remote(clear=True))
+            ray.get(signal_actor.wait.remote())
```

Comment:
I think we need to use two different signal actors otherwise if `signal_actor.send.remote(clear=True)` happens before `ray.get(signal_actor.wait.remote())`, the signal is lost.

Reply:
I removed `clear=True` from the last `send` in these two tests.

Reply:
I eventually decided to use two signal actors because I found that I still needed two actors even if I use semaphores.